### PR TITLE
[Sol->Yul] Implement inline arrays.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -310,7 +310,31 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
 bool IRGeneratorForStatements::visit(TupleExpression const& _tuple)
 {
 	if (_tuple.isInlineArray())
-		solUnimplementedAssert(false, "");
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(*_tuple.annotation().type);
```
Comment (mijovic) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]:
```suggestion
		auto const& arrayType = dynamic_cast<ArrayType const&>(*_tuple.annotation().type);
```

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -310,7 +310,31 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
...
+		{
+			Expression const& component = *_tuple.components()[i];
+			component.accept(*this);
+			IRVariable converted = convert(component, baseType);
+			m_code <<
+				m_utils.writeToMemoryFunction(baseType) <<
+				"(" <<
+				("add(" + mpos + ", " + to_string(i * arrayType.memoryStride()) + ")") <<
+				", " <<
+				converted.name() <<
```
Comment (mijovic) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:335]: maybe like this
`convert(component, baseType).name()`
and delete line 
`IRVariable converted = convert(component, baseType);`

**Review by mijovic — DISMISSED** [review]
Looks good, only two small style suggestions

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp (hunk 2 above)
Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:335]: `convert` can define a new variable, so you cannot use it inline.

Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:335]: Or at least it would not be as safe...

**Comment by chriseth** [comment]
Enabled some more tests.

**Review by mijovic — APPROVED** [review]

**Pull request merged**